src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -317,31 +320,37 @@
      * Constructor, defaults to {@link #GITHUB_URL} as the end-point, and anonymous access, does not default any
      * {@link SCMSourceTrait} behaviours.
      *
-     * @param repoOwner the repository owner.
-     * @param repository the repository name.
-     * @param repositoryUrl HTML URL for the repository. If specified, takes precedence over repoOwner and repository.
-     * @param configuredByUrl Whether to use repositoryUrl or repoOwner/repository for configuration.
+     * This was choosen as the constructor to allow users to set only the values they need and
+     * not have to pass extra null or empty values. However, this source cannot be used until
+     * either "repositoryUrl" or "repoOwner"/"repository" are set.
+     * If both combinations are set, repositoryUrl will be used.
+     *
+     * See {@link #bindResolve} for further discussion of the reasons for this design.
+     *
      * @throws IllegalArgumentException if repositoryUrl is specified but invalid.
      * @since 2.2.0
      */
-    // configuredByUrl is used to decide which radioBlock in the UI the user had selected when they submitted the form.
     @DataBoundConstructor
-    public GitHubSCMSource(String repoOwner, String repository, String repositoryUrl, boolean configuredByUrl) {
-        if (!configuredByUrl) {
-            this.apiUri = GITHUB_URL;
-            this.repoOwner = repoOwner;
-            this.repository = repository;
-            this.repositoryUrl = null;
-        } else {
-            GitHubRepositoryInfo info = GitHubRepositoryInfo.forRepositoryUrl(repositoryUrl);
-            this.apiUri = info.getApiUri();
-            this.repoOwner = info.getRepoOwner();
-            this.repository = info.getRepository();
-            this.repositoryUrl = info.getRepositoryUrl();
-        }
+    public GitHubSCMSource() {
         pullRequestMetadataCache = new ConcurrentHashMap<>();
         pullRequestContributorCache = new ConcurrentHashMap<>();
-        this.traits = new ArrayList<>();
+        traits = new ArrayList<>();
+        repoOwner = "";
+        repository = "";
+        repositoryUrl = null;
+        apiUri = GITHUB_URL;
+    }
+
+    /**
+     * Legacy constructor.
+     *
+     * @param repositoryUrl HTTP URL for the repository.
+     * @since 2.2.0
+     */
+    @Deprecated
```

Comment:
Is it really required to be deprecated or just not intended to be used for data-binding? It could be used programatically without problems.

Reply:
It could be used, but we do not want that.  Also, as far as I know, there aren't any uses for these constructors aside from data binding.

Reply:
I think it would be ok to leave it undeprecated, even if just for use in tests to avoid misuse of the data binding constructor. Same with `new GitHubSCMSource(String, String)`. If you leave it deprecated, you _will_ get community PRs that convert all of the tests to use the non-deprecated constructor, which may or may not be what you want.